django/db/backends/oracle/utils.py

```diff
@@ -10,16 +10,22 @@ class InsertVar:
     insert statement.
     """
     types = {
+        'AutoField': int,
+        'BigAutoField': int,
+        'IntegerField': int,
+        'BigIntegerField': int,
+        'PositiveSmallIntegerField': int,
+        'PositiveIntegerField': int,
         'FloatField': Database.NATIVE_FLOAT,
         'CharField': str,
         'DateTimeField': Database.TIMESTAMP,
-        'DateField': Database.DATETIME,
+        'DateField': Database.Date,
         'DecimalField': Database.NUMBER,
     }
 
     def __init__(self, field):
         internal_type = getattr(field, 'target_field', field).get_internal_type()
-        self.db_type = self.types.get(internal_type, int)
+        self.db_type = self.types.get(internal_type, str)
```

Comment:
@felixxm with the `InsertVar` actually now being used in more places, I need to make a bunch of adjustment to make this work for all scenarios.
I decided to add all integer types explicitly and use `str` as a fallback. This made more senes, since the majority of fields like `UUID` or `GenericIPAddress` work just fine with strings. I believe this will also be better suited for any third party library, that might implement it's own fields.

The test that really test all different cases here is: `serializers.test_data.SerializerDataTests`. It inserts all different field types as primary keys.

Reply:
Agreed, can we remove `CharField` from `types` and move this adjustment to a separate commit, e.g. `Refs #29444 -- Made db.backends.oracle.utils.InsertVar use str as default.`